packages/vscode-extension/src/debug/depsChecker/funcToolChecker.ts

```diff
@@ -51,191 +46,224 @@ export class FuncToolChecker implements IDepsChecker {
+        "global-func-version": `${await this.queryFuncVersionSilently("func")}`,
+      });
     }
+    if (isPortableFuncInstalled) {
+      // avoid missing this event after first installation 60 days
+      this._telemetry.sendEvent(DepsCheckerEvent.funcInstallCompleted);
+    }
+
+    return isPortableFuncInstalled || isGlobalFuncInstalled;
+  }
+
+  public async isPortableFuncInstalled(): Promise<boolean> {
+    const portableFuncVersion = await this.queryFuncVersionSilently(
+      FuncToolChecker.getPortableFuncExecPath()
+    );
+    return portableFuncVersion !== null && supportedVersions.includes(portableFuncVersion);
+  }
+
+  public async isGlobalFuncInstalled(): Promise<boolean> {
+    const globalFuncVersion = await this.queryFuncVersionSilently("func");
+    return globalFuncVersion !== null && supportedVersions.includes(globalFuncVersion);
   }
 
   public async install(): Promise<void> {
     if (!(await this.hasNPM())) {
-      // provided with Learn More link if npm doesn't exist.
-      this._telemetry.sendUserErrorEvent(
-        DepsCheckerEvent.funcInstall,
-        TelemtryMessages.NPMNotFound
-      );
-      throw new DepsCheckerError(
-        Messages.needInstallFuncCoreTool.replace("@NameVersion", installedNameWithVersion),
-        defaultHelpLink
-      );
+      this.handleNpmNotFound();
     }
 
+    await this.cleanup();
+    await this.installFunc();
+
+    if (!(await this.validate())) {
+      await this.handleInstallFuncFailed();
+    }
+
+    this._telemetry.sendEvent(DepsCheckerEvent.funcInstallCompleted);
     await this._logger.info(
-      Messages.startInstallFunctionCoreTool.replace("@NameVersion", installedNameWithVersion)
+      Messages.finishInstallFunctionCoreTool.replace("@NameVersion", displayFuncName)
     );
+  }
 
-    try {
-      await this._telemetry.sendEventWithDuration(
-        DepsCheckerEvent.funcInstallCompleted,
-        async () => {
-          await this._adapter.runWithProgressIndicator(async () => {
-            await this.installFuncCoreTools(FuncVersion.v3);
-          });
-        }
-      );
-    } catch (error) {
-      this._telemetry.sendSystemErrorEvent(
-        DepsCheckerEvent.funcInstall,
-        TelemtryMessages.failedToInstallFunc,
-        error
-      );
+  private async handleInstallFuncFailed(): Promise<void> {
+    await this.cleanup();
 
-      throw new DepsCheckerError(
-        Messages.failToInstallFuncCoreTool.replace("@NameVersion", installedNameWithVersion),
-        defaultHelpLink
-      );
-    }
+    this._telemetry.sendSystemErrorEvent(
+      DepsCheckerEvent.funcInstallError,
+      TelemtryMessages.failedToInstallFunc,
+      Messages.failToValidateFuncCoreTool.replace("@NameVersion", displayFuncName)
+    );
 
-    // validate after installation.
-    const isInstalled = await this.isInstalled();
-    if (!isInstalled) {
+    throw new DepsCheckerError(
+      Messages.failToInstallFuncCoreTool.replace("@NameVersion", displayFuncName),
```

Comment:
`replace` only replaces the first occurrence. The displayed message will be 
![image](https://user-images.githubusercontent.com/9698542/124695105-256ade00-df15-11eb-83fe-3cda0fb6c6e6.png)

Reply:
Yep, has fixed it.